Lib/fontTools/ttLib/ttGlyphSet.py

```diff
@@ -0,0 +1,214 @@
+	that it has .draw() and .drawPoints() methods that take a pen object as
+	their only argument. Additionally there are 'width' and 'lsb' attributes,
+	read from the 'hmtx' table.
+
+	If the font contains a 'vmtx' table, there will also be 'height' and 'tsb'
+	attributes.
+	"""
+
+	def __init__(self, glyphset, glyph, horizontalMetrics, verticalMetrics=None):
+		"""Construct a new _TTGlyph.
+
+		Args:
+			glyphset (_TTGlyphSet): A glyphset object used to resolve components.
+			glyph (ttLib.tables._g_l_y_f.Glyph): The glyph object.
+			horizontalMetrics (int, int): The glyph's width and left sidebearing.
+		"""
+		self._glyphset = glyphset
+		self._glyph = glyph
+		self.width, self.lsb = horizontalMetrics
+		if verticalMetrics:
+			self.height, self.tsb = verticalMetrics
+		else:
+			self.height, self.tsb = None, None
+
+	def draw(self, pen):
+		"""Draw the glyph onto ``pen``. See fontTools.pens.basePen for details
+		how that works.
+		"""
+		self._glyph.draw(pen)
+
+	def drawPoints(self, pen):
+		# drawPoints is only implemented for _TTGlyphGlyf at this time.
+		raise NotImplementedError()
+
+class _TTGlyphCFF(_TTGlyph):
+	pass
+
+class _TTGlyphGlyf(_TTGlyph):
+
+	def draw(self, pen):
+		"""Draw the glyph onto Pen. See fontTools.pens.basePen for details
+		how that works.
+		"""
+		glyfTable = self._glyphset._glyphs
+		glyph = self._glyph
+		offset = self.lsb - glyph.xMin if hasattr(glyph, "xMin") else 0
+		glyph.draw(pen, glyfTable, offset)
+
+	def drawPoints(self, pen):
+		"""Draw the glyph onto PointPen. See fontTools.pens.pointPen
+		for details how that works.
+		"""
+		glyfTable = self._glyphset._glyphs
+		glyph = self._glyph
+		offset = self.lsb - glyph.xMin if hasattr(glyph, "xMin") else 0
+		glyph.drawPoints(pen, glyfTable, offset)
+
+
+
+class _TTVarGlyphSet(object):
+
+	def __init__(self, font, location, normalized=False):
+		from fontTools.varLib.models import normalizeLocation, piecewiseLinearMap
+		self._ttFont = font
+		if not normalized:
+			axes = {a.axisTag: (a.minValue, a.defaultValue, a.maxValue) for a in font['fvar'].axes}
+			location = normalizeLocation(location, axes)
+			if 'avar' in font:
+				avar = font['avar']
+				avarSegments = avar.segments
+				new_location = {}
+				for axis_tag,value in location.items():
+					avarMapping = avarSegments.get(axis_tag, None)
+					if avarMapping is not None:
+						value = piecewiseLinearMap(value, avarMapping)
+					new_location[axis_tag] = value
+				location = new_location
+				del new_location
+
+		self.location = location
+
+	def keys(self):
+		return list(self._ttFont['glyf'].keys())
+
+	def has_key(self, glyphName):
+		return glyphName in self._ttFont['glyf']
+	__contains__ = has_key
+
+	def __getitem__(self, glyphName):
+		return _TTVarGlyphGlyf(self._ttFont, glyphName, self.location)
+
+	def get(self, glyphName, default=None):
+		try:
+			return self[glyphName]
+		except KeyError:
+			return default
+
+def setCoordinates(glyph, coord, glyfTable):
```

Comment:
looks almost identical to `glyf._setCoordinates`, ideally we'd factor out the shared code. Maybe not worth in this case. I see you don't round here, whereas glyf._setCoordinates does, but I guess that's intentional.

Reply:
I can add rounding just to be similar to the other case. It doesn't need rounding. I copied code from Just.

You are right that ideally this code should be refactored.